### PR TITLE
frontend/aopp: improve Aopp type 

### PR DESCRIPTION
**Pull request by benma**
Builds on https://github.com/digitalbitbox/bitbox-wallet-app/pull/1392 because of a conflict, otherwise independent.

### CODE REVIEW

**Review by thisconnect — APPROVED** [review]
LGTM with nit

File: frontends/web/src/api/aopp.ts [hunk 1]
```diff
@@ -26,28 +26,28 @@ interface Accounts extends Array<Account> {
...
-    // See backend/aopp.go for a description of the states.
-    state: 'inactive' | 'user-approval' | 'choosing-account' | 'signing' | 'success';
+export type Aopp = {
+    state: 'error';
+    errorCode: 'aoppUnsupportedAsset' | 'aoppVersion' | 'aoppInvalidRequest' | 'aoppNoAccounts' | 'aoppUnsupportedKeystore' | 'aoppUnknown' | 'aoppSigningAborted' | 'aoppCallback';
+    callback: string;
+} | {
+    state: 'inactive';
+} | {
+    state: 'user-approval' | 'awaiting-keystore' | 'syncing';
```
[the remark this answers is not in the document]
Comment (benma) [reply frontends/web/src/api/aopp.ts:36]: But it also has accounts which the other ones don't.

File: frontends/web/src/components/aopp/aopp.tsx [hunk 2]
```diff
@@ -73,33 +82,25 @@ class Aopp extends Component<Props, State> {
...
-
     public render(
         { t, aopp }: RenderableProps<Props>,
         { accountCode }: State,
     ) {
         if (!aopp) {
             return null;
         }
-        const domain = aopp.callback ? new URL(aopp.callback).host : '';
+        const domain = (callback: string) => new URL(callback).host;
```
[the remark this answers is not in the document]
Comment (benma) [reply frontends/web/src/components/aopp/aopp.tsx]: Moved to a function, not method, as it does not reference `this`.

File: frontends/web/src/components/aopp/aopp.tsx [hunk 3]
```diff
@@ -24,7 +24,8 @@ import { Fullscreen, FullscreenHeader, FullscreenContent, FullscreenButtons } fr
 import { Message } from '../message/message';
 import { Button, Field, Input, Label, Select } from '../forms';
 import { CopyableInput } from '../copy/Copy';
-import { ArrowDown, BitBox02Stylized, Cancel, Checked } from '../icon';
+import { ArrowDown, BitBox02Stylized,Cancel, Checked } from '../icon';
```
[the remark this answers is not in the document]
Comment (benma) [reply frontends/web/src/components/aopp/aopp.tsx]: done

File: frontends/web/src/api/aopp.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (thisconnect) [reply frontends/web/src/api/aopp.ts:36]: I was blind, 👍

**Pull request merged**